src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs

```diff
@@ -67,20 +85,20 @@ public LinuxAppServiceEventGenerator(LinuxAppServiceFileLoggerFactory loggerFact
         public override void LogFunctionExecutionEvent(string executionId, string siteName, int concurrency, string functionName,
             string invocationId, string executionStage, long executionTimeSpan, bool success)
         {
-            var logger = _loggerFactory.GetOrCreate(FunctionsExecutionEventsCategory);
             string currentUtcTime = DateTime.UtcNow.ToString();
-            if (FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableLinuxEPExecutionCount))
+            bool detailedExecutionEventsDisabled = _functionsHostingConfigOptions.Value.DisableLinuxAppServiceExecutionDetails;
```

Comment:
nit: Once resolved, this value won't changed, so storing this result would be more efficient here as it would avoid the overhead of the method calls and dictionary lookup.

This assuming we're OK with this not changing after initialization. If we want to react to hosting configuration changes, we'd need to change from `IOptions<FunctionsHostingConfigOptions>` to `IOptionsMonitor<FunctionsHostingConfigOptions>`

Reply:
Thanks @fabiocav ! Tagging @CooperLink to address this and 

> Outside of the scope of this PR, but it would be good to consider an enhancement to the logger so it's more efficient and auto-detects when it needs to process events.

in a separate PR!